distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -101,9 +101,20 @@ type Table
     at : Text | Integer -> Column ! No_Such_Column | Index_Out_Of_Bounds
     at self selector=0 = case selector of
         _ : Integer -> self.make_column (self.internal_columns.at selector)
-        _ : Text ->
-            internal_column = self.internal_columns.find if_missing=(Error.throw (No_Such_Column.Error selector)) (p -> p.name == selector)
-            self.make_column internal_column
+        _ -> self.get selector (Error.throw (No_Such_Column.Error selector))
+
+    ## Returns the column with the given name or index.
+
+       Arguments:
+       - selector: The name or index of the column being looked up.
+       - if_missing: The value to use if the selector isn't present.
+    get : Text | Integer -> Any -> Column
+    get self selector=0 ~if_missing=Nothing =
+        internal_column = case selector of
+            _ : Integer -> self.internal_columns.get selector if_missing=Nothing
+            _ : Text -> self.internal_columns.find (p -> p.name == selector) if_missing=Nothing
+            _ -> Error.throw (Type_Error.Error Text selector "selector")
```

Comment:
The valid types are either text OR int, here we show just int. I know that we cannot construct a sum type at value level (we really should be at some point...), so for now maybe we can (ab)use the type error format and do:
```suggestion
            _ -> Error.throw (Type_Error.Error "Text | Integer" selector "selector")
```
I think this should show the correct message then (although worth double checking in tests by `errored_expr.catch.to_display_text.should_equal "..."`). I think it's important to show the correct expected type to the user within the error. 

If we cannot use type error, maybe let's fall back to `Illegal_Argument`, as long as we can provide an informed explanation.